static/js/analytics.js

```diff
@@ -265,26 +265,35 @@ const renderChartsFromSingleUserJson = (json) => {
   renderTaggedFriendsGraphFromJson(json);
 };
 
-const getAndRenderAnalytics = async (startDateTimestamp, endDateTimestamp) => {
-  if (!startDateTimestamp && !endDateTimestamp) {
-    // Default date interval is from 3 months ago
-    const monthsAgo = 3;
-    const currentDate = new Date();
-    const startDate = new Date(
-      new Date().setMonth(currentDate.getMonth() - monthsAgo)
-    );
-    endDateTimestamp = Math.trunc(currentDate.getTime() / 1000);
-    startDateTimestamp = Math.trunc(
-      new Date().setMonth(currentDate.getMonth() - monthsAgo) / 1000
-    );
-    // Setting default values on the date picker
-    document.getElementById("start-date").value = startDate
-      .toISOString()
-      .split("T")[0];
-    document.getElementById("end-date").value = currentDate
-      .toISOString()
-      .split("T")[0];
-  }
+const updateDatePicker = (startDate, endDate) => {
+  document.getElementById("start-date").value = startDate;
+  document.getElementById("end-date").value = endDate;
+};
+
+const getDateIntervalAndUpdateDatePicker = () => {
+  const startDateField = document.getElementById("start-date").value;
+  const endDateField = document.getElementById("end-date").value;
+  const monthsAgo = 3;
+  let endDate = endDateField ? new Date(endDateField) : new Date();
+  let startDate = startDateField
+    ? new Date(startDateField)
+    : new Date(new Date().setMonth(endDate.getMonth() - monthsAgo));
+
+  endDate.setDate(endDate.getDate() + 1); // Handles the last day
```

Comment:
If every time this get called you update the last day with +1 you can keep going forward in time. Instead move this part of the logic to the comparison function on the Python API side.

Reply:
Moved logic on backend